browser/ui/brave_tooltips/brave_tooltip_popup.h

```diff
@@ -0,0 +1,155 @@
+#include <string>
+
+#include "base/scoped_observation.h"
+#include "brave/browser/ui/brave_tooltips/brave_tooltip.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/display/display_observer.h"
+#include "ui/gfx/animation/animation_delegate.h"
+#include "ui/gfx/shadow_util.h"
+#include "ui/gfx/shadow_value.h"
+#include "ui/views/widget/widget_delegate.h"
+#include "ui/views/widget/widget_observer.h"
+
+class Profile;
+
+namespace gfx {
+class LinearAnimation;
+class Point;
+class Rect;
+class Size;
+}  // namespace gfx
+
+namespace views {
+class Widget;
+}  // namespace views
+
+namespace brave_tooltips {
+
+class BraveTooltipView;
+
+// The widget delegate of a tooltip popup. The view is owned by
+// the widget
+class BraveTooltipPopup : public views::WidgetDelegateView,
+                          public views::WidgetObserver,
+                          public gfx::AnimationDelegate,
+                          public display::DisplayObserver {
+ public:
+  METADATA_HEADER(BraveTooltipPopup);
+
+  BraveTooltipPopup(Profile* profile, std::unique_ptr<BraveTooltip> tooltip);
+  ~BraveTooltipPopup() override;
+
+  // Show the tooltip popup view for the given |profile| and
+  // |tooltip|
+  static void Show(Profile* profile, std::unique_ptr<BraveTooltip> tooltip);
+
+  // Close the tooltip popup view for the given |tooltip_id|.
+  // |by_user| is true if the tooltip popup was closed by the user,
+  // otherwise false
+  static void Close(const std::string& tooltip_id, const bool by_user);
+
+  // Close the widget for the given |tooltip_id|
+  static void CloseWidget(const std::string& tooltip_id);
+
+  // User pressed the Ok button for the given |tooltip_id|
+  static void OnOkButtonPressed(const std::string& tooltip_id);
+
+  // User pressed the Cancel button for the given |tooltip_id|
+  static void OnCancelButtonPressed(const std::string& tooltip_id);
+
+  void set_normalized_display_coordinates(double x, double y);
+
+  void set_display_work_area_insets(int x, int y);
+
+  // display::DisplayObserver:
+  void OnDisplayRemoved(const display::Display& old_display) override;
+  void OnDisplayMetricsChanged(const display::Display& display,
+                               uint32_t changed_metrics) override;
+
+  // views::WidgetDelegateView:
+  void GetAccessibleNodeData(ui::AXNodeData* node_data) override;
+  void OnDisplayChanged() override;
+  void OnWorkAreaChanged() override;
+  void OnPaintBackground(gfx::Canvas* canvas) override;
+  void OnThemeChanged() override;
+
+  // views::WidgetObserver:
+  void OnWidgetCreated(views::Widget* widget) override;
+  void OnWidgetDestroyed(views::Widget* widget) override;
+
+  // AnimationDelegate:
+  void AnimationEnded(const gfx::Animation* animation) override;
+  void AnimationProgressed(const gfx::Animation* animation) override;
+  void AnimationCanceled(const gfx::Animation* animation) override;
+
+ private:
+  enum class AnimationState {
+    // No animation is running
```

Comment:
Nit: looks like the enum names are self-documenting, so comments aren't strictly necessary.

Reply:
I agree! The comment were copied from Chromium animation code

Reply:
Fixed.